### PR TITLE
Add direct references to non-CRW available plugins

### LINKED ISSUE #18580: Che docs devfile is broken
### Describe the bug
It is not possible to start a workspace form [che-docs](https://github.com/eclipse/che-docs) repository [devfile](https://github.com/eclipse/che-docs/blob/master/devfile.yaml)

### Che version
- nightly
- latest

### Steps to reproduce
1. Open Che dashboard
2. Go to `Get Started` tab
3. Switch to `Custom Workspace`
4. Insert `https://github.com/eclipse/che-docs` into devfile url field
5. Create and try to start the workspace

### Expected behavior
It is possible to start a workspace by providing docs repository link while creating workspace from Che dasbboard

### Runtime
- Openshift (hosted Che)
- Minikube

### Screenshots
Screenshot from 2020-12-09 17-17-11

### Installation method
chectl

### Environment
- Hosted Che
- Minikube locally

### Eclipse Che Logs
Che server logs has no errors

### Additional context
Che server shows that workspace is started, but Theia fails to connect to Che server (at least it seems so because of yellow status bar).
When look at workspace details, there is a warning about deprecated plugins.

### PR DESCRIPTION
**Pull request by ericwill**
Signed-off-by: Eric Williams <ericwill@redhat.com>

> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Add direct links to the nightly plugin definitions of:
* vale plugin
* asciidoc plugin
* shellcheck plugin

### What issues does this PR fix or reference?
Fixes eclipse/che#18580

### Specify the version of the product this PR applies to.
N/A

### PR Checklist

As the author of this Pull Request I made sure that:

- [ ] `vale` has been run successfully against the PR branch
- [ ] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

**Comment by themr0c** [comment]
That's nice. I will test on Monday.

**Comment by themr0c** [comment]
I received the following error at first start. The second start was fine:

```
Press F5 or click here to try again in debug mode. Download logs

Processing plugin timonwong/shellcheck/latest
  Installing plugin extension 1/1
    Downloading plugin from https://github.com/timonwong/vscode-shellcheck/releases/download/v0.13.2/shellcheck-0.13.2.vsix

Saving log of installed plugins

All plugin artifacts have been successfully downloaded

Pulling image "registry.redhat.io/codeready-workspaces/jwtproxy-rhel8@sha256:7c5a5bd48f5582d730323fff61b617b887f9c2320e1b0788c40847175e9d3fd0"

Successfully pulled image "registry.redhat.io/codeready-workspaces/jwtproxy-rhel8@sha256:7c5a5bd48f5582d730323fff61b617b887f9c2320e1b0788c40847175e9d3fd0" in 1.101357664s

Created container che-jwtproxy5814soa7

Started container che-jwtproxy5814soa7

Pulling image "quay.io/eclipse/che-docs:latest"

Successfully pulled image "quay.io/eclipse/che-docs:latest" in 6.553491228s

Created container che-docs

Started container che-docs

Pulling image "quay.io/eclipse/che-plugin-sidecar@sha256:f25f9b8eef7e2a7aea3e842ee13626935b98f96327f5c56b6bbf48b49754c45d"

Successfully pulled image "quay.io/eclipse/che-plugin-sidecar@sha256:f25f9b8eef7e2a7aea3e842ee13626935b98f96327f5c56b6bbf48b49754c45d" in 2.140880353s

Created container vscode-valescx

Started container vscode-valescx

Pulling image "quay.io/eclipse/che-plugin-sidecar@sha256:3a5f128b217625c211f69fd2277300ae611fcf7da287c2786e522f13a81ed701"

Successfully pulled image "quay.io/eclipse/che-plugin-sidecar@sha256:3a5f128b217625c211f69fd2277300ae611fcf7da287c2786e522f13a81ed701" in 11.939882287s

Created container asciidoctor-vscodedbw

Started container asciidoctor-vscodedbw

Pulling image "quay.io/eclipse/che-plugin-sidecar@sha256:6b9c739f487738a5bfb20e052a21b4edff4289c738496708abf8adc4d5820d79"

Error: Failed to run the workspace: "The following containers have terminated:
shellcheckdeb: reason = 'OOMKilled', exit code = 137, message = 'null'"

```

**Comment by themr0c** [comment]
ShellCheck plug-in works as expected. 

Vale plug-ins not fully funtional: 

* The option "Use Vale's CLI instead of Vale Server. " should be enabled to enable reporting in the "Problems" view. But even with the option enabled, no errors appear in the "Problems" view when opening a file known to have vale errors (running vale from a the vale container shell shows the errors).

* When opening a shell in the plugin container, running vale agains all che-docs files fails with following error:
```
vale .
Killed
```

**Comment by themr0c** [comment]
I spotted a missing dependency in che-docs container while running tests:  https://github.com/eclipse/che/issues/19228

**Comment by nickboldt** [comment]
> shellcheckdeb: reason = 'OOMKilled', exit code = 137, message = 'null'"

If this is anything like the problems we had with theia and several plugins on ppc64le, solution is to grant more `memoryLimit` to the sidecars / plugins. 

https://github.com/eclipse/che-docs/pull/1889/files#diff-23d39e3ae288b080d8aa0995928cd3c432f887a75f0ab3bba675089f51a4af8eR10

**Comment by themr0c** [comment]
> grant more ram to the sidecars / plugins.

In the devfile, or in the defaults for the plugin?

**Comment by nickboldt** [comment]
Also should `https://github.com/timonwong/vscode-shellcheck/releases/download/v0.13.2/shellcheck-0.13.2.vsix` be pulled from a RH-built location, eg., https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/vf278912/vscode-markdown-mermaid-1.9.2.vsix

**Comment by nickboldt** [comment]
> In the devfile, or in the defaults for the plugin?

See https://issues.redhat.com/browse/CRW-1475 -- quick workaround is to add more in the devfile, but we ultimately added more in the plugins themselves so the fix is more globally available to anyone using a plugin on that arch.

**Comment by ericwill** [comment]
I also experienced the `OOMKilled` issue on the first run of the shellcheck plugin. Restarting the workspace solved it and I didn't see the error again, not sure what's going on there.

I can enable the setting for `Use Vale's CLI instead of Vale Server` in the devfile. FWIW all the issues you are experiencing were there in the old devfile too, it's using the same plugins.


> Also should `https://github.com/timonwong/vscode-shellcheck/releases/download/v0.13.2/shellcheck-0.13.2.vsix` be pulled from a RH-built location, eg., https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/vf278912/vscode-markdown-mermaid-1.9.2.vsix

This is an upstream devfile not shipped with CRW, so I think we're okay.

**Comment by nickboldt** [comment]
> This is an upstream devfile not shipped with CRW, so I think we're okay.

True... until someone wants to use this plugin w/ workspaces.openshift.com and then complains that it doesn't work (or worse, it introduces a security flaw). 

Sure, it's not supported there officially, but would still be better to be building everything that will one day be used in the product on RH hardware w/ RH process so we have better control of source / CVE fixes.

**Comment by ericwill** [comment]
Alright, I'll build those plugins and change the links upstream -- stay tuned.

**Review by themr0c — APPROVED** [review]

**Comment by ericwill** [comment]
All changes to this devfile are finished, so it can be merged. Memory limits and vsix links will be updated in the plugin registry.

**Review by benoitf — APPROVED** [review]

File: devfile.yaml [hunk 1]
```diff
@@ -21,14 +21,19 @@ components:
         port: 4000
         attributes:
           path: /che-7/overview/introduction-to-eclipse-che/
-  - id: errata-ai/vale-server/latest
-    type: chePlugin
+  - type: chePlugin
+    reference: https://che-plugin-registry-main.surge.sh/v3/plugins/errata-ai/vale-server/latest/meta.yaml
```
Comment (benoitf) [inline devfile.yaml:25]: It could be safer if we had deployment per release but at least here we ensure to use fresh stuff for dogfooding :-)

**Pull request merged**